### PR TITLE
[DOCS] D/dx 237/tal docstrings

### PR DESCRIPTION
**Pull request by talagluck**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- Update docstrings for 3 methods

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py [hunk 1]
```diff
@@ -194,9 +194,13 @@ def show_expectations_by_expectation_type(
         include_profiler_config: bool = False,
         send_usage_event: bool = True,
     ) -> None:
-        """
-        Populates named "ExpectationSuite" with "ExpectationConfiguration" list, stored in "DataAssistantResult" object,
-        and displays this "ExpectationConfiguration" list, grouped by "expectation_type", in predetermined order.
+        """Populates named "ExpectationSuite" with "ExpectationConfiguration" list stored in "DataAssistantResult"
+        object and displays this "ExpectationConfiguration" list, grouped by "expectation_type", in predetermined order.
```
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py]: Can we shorten this? And backticks will probably render better than quotes around entity names

File: great_expectations/rule_based_profiler/rule_based_profiler.py [hunk 2]
```diff
@@ -1328,6 +1328,11 @@ def rule_states(self) -> List[RuleState]:
         return self._rule_states
 
     def to_json_dict(self) -> dict:
```
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/rule_based_profiler.py:1334]: missing decorator

File: great_expectations/validator/metric_configuration.py [hunk 3]
```diff
@@ -138,6 +138,12 @@ def id(self) -> Tuple[str, str, str]:
         )
 
     def to_json_dict(self) -> dict:
```
Comment (anthonyburdi) [inline great_expectations/validator/metric_configuration.py:143]: missing decorator

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
just a few nits

**Comment by talagluck** [comment]
Thanks, updated!

**Review by anthonyburdi — APPROVED** [review]

**Pull request merged**